tests/Aspire.EndToEnd.Tests/ProjectInfo.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+
+namespace Aspire.EndToEnd.Tests;
+
+public sealed class ProjectInfo
+{
+    [JsonIgnore]
+    public HttpClient Client { get; set; } = default!;
+
+    public EndpointInfo[] Endpoints { get; set; } = default!;
+
+    /// <summary>
+    /// Sends a GET request to the specified resource and returns the response message.
+    /// </summary>
+    public Task<HttpResponseMessage> HttpGetAsync(string bindingName, string path, CancellationToken cancellationToken = default)
+    {
+        var allocatedEndpoint = Endpoints.Single(e => e.Name == bindingName);
+        var url = $"{allocatedEndpoint.Uri}{path}";
+
+        return Client.GetAsync(url, cancellationToken);
+    }
+
+    /// <summary>
+    /// Sends a GET request to the specified resource and returns the response body as a string.
+    /// </summary>
+    public Task<string> HttpGetStringAsync(string bindingName, string path, CancellationToken cancellationToken = default)
+    {
+        var allocatedEndpoint = Endpoints.Single(e => e.Name == bindingName);
+        var url = $"{allocatedEndpoint.Uri}{path}";
+
+        return Client.GetStringAsync(url, cancellationToken);
+    }
+
+    public async Task WaitForHealthyStatusAsync(string bindingName, CancellationToken cancellationToken = default)
+    {
+        while (true)
+        {
+            try
+            {
+                await HttpGetStringAsync(bindingName, "/health", cancellationToken);
```

Comment:
Should this check the returned string? I'm assuming that this can return one of https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.diagnostics.healthchecks.healthstatus?view=dotnet-plat-ext-8.0 .

Reply:
Done.